src/variables/variable_unit_flow.jl

```diff
@@ -139,7 +139,9 @@ function add_variable_unit_flow!(m::Model)
             ),
         )
         for (u, n_ref, d_ref, n, d, fix_ratio, direct) in _related_flows(fix_ratio_d1_d2)
-        if _has_simple_fix_ratio_unit_flow(n, n_ref, fix_ratio)
+        if _has_simple_fix_ratio_unit_flow(n, n_ref, fix_ratio) && 
+            isempty(unit_flow_op_indices(m; unit=u, node=n_ref, direction=d_ref)) && 
+            isempty(unit_flow_op_indices(m; unit=u, node=n, direction=d))
```

Comment:
Again, the two new checks should go inside `_has_simple_fix_ratio_unit_flow` so we don't need to duplicate the code here and in the fix_ratio constraint.